packages/tools/devtools/devtools-browser-extension/src/devtools/BackgroundConnection.ts

```diff
@@ -167,10 +142,43 @@ export class BackgroundConnection
 	 * Immediately throws, since this type is currently not capable of recovering from this state.
 	 */
 	private readonly onBackgroundServiceDisconnect = (): void => {
-		throw new Error(
+		console.log(
 			formatDevtoolsScriptMessageForLogging(
-				"The Background Script disconnected. Further use of the message relay is not allowed.",
+				"Disconnected from Background script. Re-connecting again..",
 			),
 		);
+		this.connectToBackgroundService();
+	};
+
+	/**
+	 * Connects to the Background Script.
+	 */
+	private readonly connectToBackgroundService = (): void => {
```

Comment:
For reference: I was slightly concerned that not de-registering the event handlers from the previous `backgroundServiceConnection` might be a memory leak, but did some research and it seems like that's not the case, the event emitter will be GCed if it cannot be reached regardless of listeners attached to it. So we should be ok.

Reply:
Awesome! Thanks for confirming it. You are the best

Reply:
That's good to know! I assumed it as a leak as well.

It would probably be worth adding a comment to the onDisconnect handler explaining that, in case anyone in the future wonders why we aren't doing cleanup.

Reply:
Updating the doc here.